viewing_party/party.py

```diff
@@ -0,0 +1,254 @@
+
+    # return the list of movies only the user has watched
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    # get a list of the user's watched movies
+    users_movies = get_users_watched_movies(user_data)
+    # get a list of the user's friend's watched movies
+    friends_movies = get_friends_watched_movies(user_data)
+
+    # get the difference between the user's friend's movies and the user's movies
+    unique_movies = friends_movies.difference(users_movies)
+    
+    friends_unique_movies = []
+
+    # loop over the list of unique movies the user's friends have watched
+    for movie in unique_movies:
+        movie_title = {"title": movie}
+        # add the movie title to the list of unique movies
+        friends_unique_movies.append(movie_title)
+
+    # return the list of movies the user's friends have watched but the user has not
+    return friends_unique_movies
+
+def get_friends_unique_watched_with_details(user_data):
+
+    # get a list of movie only theuser's friends have watched
+    users_unwatched_movies = get_friends_unique_watched(user_data)
+    friends_list = user_data["friends"]
+
+    unwatched_movies_and_details = set()
+
+    # loop over the list of movies only the user's friends have watched
+    for unwatched_movie in users_unwatched_movies:
+        unwatched_movie_title = unwatched_movie["title"]
+        # loop over the list of the user's friends
+        for friend in friends_list:
+            friends_watched_list = friend["watched"]
+            # loop over the list of that friend's watched movies
+            for movie in friends_watched_list:
+                friends_movie_title = movie["title"]
+                # if a movie from user's unwatched movie list is in a friend's watched movies list
+                #  add the movie title and details (such as host or genre) to a list of user's unwatched movies
+                if unwatched_movie_title == friends_movie_title:
+                    movie_details = tuple(movie.values())
+                    unwatched_movies_and_details.add(movie_details)
+    
+    # return the list of movies that only the user's friends have watched including relevant details
+    return unwatched_movies_and_details
+
+def get_available_recs(user_data):
+    
+    # get a list of movies and their streaming services only the user's friends have watched
+    unwatched_movies_and_hosts = get_friends_unique_watched_with_details(user_data)
+    
+    available_recommended_movies = []
+
+    # loop over the user's list of unwatched movies and their streaming services 
+    for title, host in unwatched_movies_and_hosts:
+        # if the unwatched movie is hosted on a service that the user is subscribed to,
+        #  append the movie and service details to the list of recommended movies
+        if host in user_data["subscriptions"]:
+            available_recommended_movies.append({"title": title, "host": host})
```

Comment:
I see this happens a few more times in the project, so I promise this is the last time I'll bring it up! 😄  But it would be nice if we could make use of the movie dictionaries already in user_data rather than needing to rebuild them here.